signup-service/src/main/java/io/mosip/signup/util/IAMTokenService.java

```diff
@@ -0,0 +1,63 @@
+package io.mosip.signup.util;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+import java.time.Instant;
+import java.util.Map;
+
+@Service
+public class IAMTokenService {
+
+    private final RestTemplate restTemplate = new RestTemplate();
+    private String cachedToken;
+    private Instant expiryTime;
```

Comment:
Can we use the existing cache (simple / redis) with TTL support to store the authtoken?